ThaDeveloper/weConnect
# Challenge3

**What is in this PR?**

This is Version 2 of the weConnect API
**_Goal_**: Modify and secure the API with Postgres

_What can users do?_
Create an account and login to the API
Reset their password.
Reguster a new business
View all available businesses
View a specific business profile
Review a business
Update their business profile.
Delete their business
Search for businesses by name
Filter busines by location or category

**Testing the API**

- [ ]  Ensure you have python3 and Postman installed
- [ ] Enter weConnect directory $ cd weConnect
- [ ] Install the dependencies : $ pip install -r requirements.txt
- [ ] Create a Postgresql db and set the URI environment
     variable
- [ ] Update migrations `$ python3 manage.py upgrade`
- [ ] Run `$ flask run`

_Endpoints to be tested on Postman_

Auth
- POST ```/api/v2/auth/register```
- POST ```/api/v2/auth/login```
- POST ```/api/v2/auth/reset-password```
- DELETE ```/api/v2/auth/logout```

Business and Reviews
- POST ```/api/v2/businesses``` - Auth needed
- PUT ```/api/v2/businesses/<business_id>``` - Auth needed
- DELETE ```/api/v2/businesses/<business_id>``` - Auth needed
- GET ```/api/v2/businesses```
- GET ```/api/v2/businesses/<busines_id>```
- GET ```/api/v2/businesses/?q=<query>``` - Search a business
- GET ```/api/v2/businesses/?q=<query>&category=<category>``` - Search business in a specific category
- GET ```/api/v2/businesses/?q=<query>&location=<location>``` - Search business in specific location
- GET ```/api/v2/businesses/?location=<location>&category=<category>``` - Filter businesses
- GET ```/api/v2/businesses/?page=3&limit=5``` - Pagination and page limit
- POST ``` /api/v2/businesses/<business_id>/reviews``` - Auth needed
- GET ```/api/v2/businesses/<business_id>/reviews```
